spec/features/newflow/student_login_flow_spec.rb

```diff
@@ -139,6 +139,41 @@
         expect(page).to have_text('This is another banner.')
         screenshot!
       end
+
+      context 'when user clicks X icon' do
+        context "when user arrived from oauth app" do
+          let(:app) do
+            app = FactoryBot.create(:doorkeeper_application, skip_terms: true,
```

Comment:
would be nice to see this broken up a bit into lets and befores.

Reply:
Fixed.